packages/consumption/src/modules/notifications/itemProcessors/attributeDeleted/OwnSharedAttributeDeletedByOwnerNotificationItemProcessor.ts

```diff
@@ -0,0 +1,82 @@
+import { ValidationResult } from "../../../common";
+import { LocalNotification } from "../../local/LocalNotification";
+import { AbstractNotificationItemProcessor } from "../AbstractNotificationItemProcessor";
+
+export class OwnSharedAttributeDeletedByOwnerNotificationItemProcessor extends AbstractNotificationItemProcessor<OwnSharedAttributeDeletedByOwnerNotificationItem> {
+    private readonly _logger: ILogger;
+
+    public constructor(consumptionController: ConsumptionController) {
+        super(consumptionController);
+        this._logger = TransportLoggerFactory.getLogger(OwnSharedAttributeDeletedByOwnerNotificationItemProcessor);
+    }
+
+    public override async checkPrerequisitesOfIncomingNotificationItem(
+        notificationItem: OwnSharedAttributeDeletedByOwnerNotificationItem,
+        notification: LocalNotification
+    ): Promise<ValidationResult> {
+        const attribute = await this.consumptionController.attributes.getLocalAttribute(notificationItem.attributeId);
+
+        if (typeof attribute === "undefined") {
+            return ValidationResult.success();
+        }
+
+        if (!attribute.isPeerSharedAttribute()) {
+            return ValidationResult.error(CoreErrors.attributes.isNotPeerSharedAttribute(notificationItem.attributeId));
+        }
+
+        if (!notification.peer.equals(attribute.shareInfo.peer)) {
+            return ValidationResult.error(CoreErrors.attributes.senderIsNotPeerOfSharedAttribute(notification.peer, notificationItem.attributeId));
+        }
+
+        return ValidationResult.success();
+    }
+
+    public override async process(
+        notificationItem: OwnSharedAttributeDeletedByOwnerNotificationItem,
+        _notification: LocalNotification
+    ): Promise<OwnSharedAttributeDeletedByOwnerEvent | void> {
+        const attribute = await this.consumptionController.attributes.getLocalAttribute(notificationItem.attributeId);
+        if (typeof attribute === "undefined") return;
+
+        const deletionInfo = LocalAttributeDeletionInfo.from({
+            deletionStatus: DeletionStatus.DeletedByOwner,
+            deletionDate: CoreDate.utc()
+        });
+
+        let updatedAttribute = attribute;
+        if (typeof attribute.deletionInfo === "undefined") {
+            attribute.setDeletionInfo(deletionInfo, this.accountController.identity.address);
+            updatedAttribute = await this.consumptionController.attributes.updateAttributeUnsafe(attribute);
+        }
+
+        const predecessors = await this.consumptionController.attributes.getPredecessorsOfAttribute(attribute.id);
+        for (const predecessor of predecessors) {
+            if (typeof predecessor.deletionInfo === "undefined") {
+                predecessor.setDeletionInfo(deletionInfo, this.accountController.identity.address);
+                await this.consumptionController.attributes.updateAttributeUnsafe(predecessor);
+            }
+        }
```

Comment:
You could use a single loop over `[attribute, ...predecessors]`. This would get rid of repetition and would communicate more clearly that you're doing the same thing to the attribute and all of its predecessors.

Reply:
I thought about doing so, but I need `updatedAttribute` for the event that is returned and I thought this way it is cleaner than overwriting the variable all the time or filling an array, if I only need one entry, anyway.

Reply:
You were right, I can just use `attribute` further on and don't need `updatedAttribute`.